exchange/bitswap/bitswap.go

```diff
@@ -227,21 +227,40 @@ func (bs *Bitswap) HasBlock(ctx context.Context, blk *blocks.Block) error {
 func (bs *Bitswap) sendWantlistMsgToPeers(ctx context.Context, m bsmsg.BitSwapMessage, peers <-chan peer.ID) error {
 	set := pset.New()
 	wg := sync.WaitGroup{}
-	for peerToQuery := range peers {
 
-		if !set.TryAdd(peerToQuery) { //Do once per peer
-			continue
-		}
+loop:
+	for {
+		select {
+		case peerToQuery, ok := <-peers:
+			if !ok {
+				break loop
+			}
 
-		wg.Add(1)
-		go func(p peer.ID) {
-			defer wg.Done()
-			if err := bs.send(ctx, p, m); err != nil {
-				log.Debug(err) // TODO remove if too verbose
+			if !set.TryAdd(peerToQuery) { //Do once per peer
```

Comment:
may be worth renaming `set.TryAdd` to `set.AddOnce(peer)` or `set.AddNew(peer)` or something. `set.TryAdd(peer)` isn't the clearest thing ever.

maybe another way to do it is:

```
// say set.Add(peer) returns an int of the number of times peer has been added

if set.Add(peer) == 1 {
  // do once per peer.
}
```